### PR TITLE
Added CO2Emission Results

### PR DESCRIPTION
**Pull request by DePauwJosh**
connects #4962

### CODE REVIEW

**Comment by nbintertech** [comment]
Co2 savings results aren't working in results panel and report - also has an error. Co2Savings data are also not saving state in setup or assessment. Let me know when the functionality is good to go and I'll review

File: src/app/fsat/fsat-report/results-summary/results-summary.component.html [hunk 1]
```diff
@@ -135,7 +135,38 @@ <h4 class="show-print">Report Results</h4>
...
+        </th>
+      </tr>
+      <tr class="callout">
+        <th>Annual CO2 Emissions Savings (tonne CO<sub>2</sub>)</th>
+        <th [ngClass]="{'selected-modification': selectedModificationIndex == -1}">&mdash;</th>
+        <th *ngFor="let modification of fsat.modifications; let index = index;"
+          [ngClass]="{'positive-savings': modification.fsat.outputs.percent_annual_savings > 10, 'selected-modification': index == selectedModificationIndex}">
+          <span
+            *ngIf="modification.fsat.whatIfScenario == true">
+            {{modification.fsat.outputs.annualSavings | sigFigs:'3'}}</span>
```
Comment (rmroot) [inline src/app/fsat/fsat-report/results-summary/results-summary.component.html:162]: This should be emissions savings not annual savings..

baseline co2 emissions - modificaiton co2 emissions

File: src/app/fsat/explore-opportunities/explore-opportunities.component.ts [hunk 2]
```diff
@@ -31,7 +31,7 @@ export class ExploreOpportunitiesComponent implements OnInit {
 
   @ViewChild('resultTabs', { static: false }) resultTabs: ElementRef;
 
-
+  co2EmissionsSavings: number = 0;
```
Comment (rmroot) [inline src/app/fsat/explore-opportunities/explore-opportunities.component.ts:34]: unused remove

File: src/app/fsat/fsat-report/fsat-report.component.ts [hunk 3]
```diff
@@ -39,6 +40,8 @@ export class FsatReportComponent implements OnInit {
   @ViewChild('reportBtns', { static: false }) reportBtns: ElementRef;
   @ViewChild('reportHeader', { static: false }) reportHeader: ElementRef;
 
+ 
+  co2EmissionsSavings: number = 0;
```
Comment (rmroot) [inline src/app/fsat/fsat-report/fsat-report.component.ts:44]: unused remove

Comment (rmroot) [reply src/app/fsat/fsat-report/fsat-report.component.ts:44]: also remove unused added imports

File: src/app/fsat/fsat-report/results-summary/results-summary.component.html [hunk 4]
```diff
@@ -135,7 +135,38 @@ <h4 class="show-print">Report Results</h4>
           [ngClass]="{'selected-modification': index == selectedModificationIndex}">
           {{modification.fsat.outputs.motorPower
           | number:'1.0-1'}}</td>
+      </tr>  
+      
+      
+
+    <!-- ^^^ Co2 section should mimick how annual energy and energy savings display their data below. 
+      You'll need to set these values in the fsat-report since there's no get diff here-->
```
Comment (rmroot) [inline src/app/fsat/fsat-report/results-summary/results-summary.component.html:143]: Should remove comments

File: src/app/fsat/fsat-results/fsat-results-panel/fsat-results-panel.component.ts [hunk 5]
```diff
@@ -18,12 +18,16 @@ export class FsatResultsPanelComponent implements OnInit {
...
 
   @HostListener('window:resize', ['$event'])
   onResize(event) {
     this.hideResults();
   }
-
+  fsatResults: ExploreOpportunitiesResults;
+  co2EmissionsSavings: number = 0;
+  annualSavings: number = 0;
+  percentSavings: number = 0;
```
Comment (rmroot) [inline src/app/fsat/fsat-results/fsat-results-panel/fsat-results-panel.component.ts:30]: remove unused code in this component as well

File: src/app/fsat/fsat.service.ts [hunk 6]
```diff
@@ -12,14 +12,16 @@ import { FanEfficiencyInputs } from '../calculator/fans/fan-efficiency/fan-effic
...
 
 
 declare var fanAddon: any;
 
 @Injectable()
 export class FsatService {
-
-
+  fsatOperations: FsatOperations;
+  fsatInputs: FsatInput;
```
Comment (rmroot) [inline src/app/fsat/fsat.service.ts:24]: cleanup unused additions in this service

File: src/app/fsat/operations/operations.component.ts [hunk 7]
```diff
@@ -37,6 +39,11 @@ export class OperationsComponent implements OnInit {
   onResize(event) {
     this.setOpHoursModalWidth();
   }
+  co2SavingsFormDisabled: boolean;
+
+  cO2SavingsData: Co2SavingsData;
+
+  fsatInputs: FsatInput;
```
Comment (rmroot) [inline src/app/fsat/operations/operations.component.ts:46]: unused remove

File: src/app/fsat/operations/operations.service.ts [hunk 8]
```diff
@@ -40,5 +40,9 @@ export class OperationsService {
     let form: FormGroup = this.getFormFromObj(obj);
     return form.valid;
   }
+  
+  // getFsatInputsFromForm(form: FormGroup, fsatInputs: FsatInput): FsatInput {
```
Comment (rmroot) [inline src/app/fsat/operations/operations.service.ts:44]: remove commented code

File: src/app/shared/models/fans.ts [hunk 9]
```diff
@@ -21,12 +23,15 @@ export interface FSAT {
   modalFieldData?: FieldData;
   existingDataUnits?: string;
   whatIfScenario?: boolean;
+  cO2SavingsData?: Co2SavingsData;
+  input?: FsatInput;
```
Comment (rmroot) [inline src/app/shared/models/fans.ts:27]: remove input addition

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/fsat/fsat.service.ts [hunk 10]
```diff
@@ -173,6 +171,7 @@ export class FsatService {
 
   //fsat results
   getResults(fsat: FSAT, isBaseline: boolean, settings: Settings): FsatOutput {
+    let co2EmissionsOutput: number;
```
Comment (rmroot) [inline src/app/fsat/fsat.service.ts:174]: unused

File: src/app/fsat/operations/operations.component.ts [hunk 11]
```diff
@@ -78,23 +83,27 @@ export class OperationsComponent implements OnInit {
 
 
   init() {
+    this.setCo2SavingsData();
     if (this.fsatOperations) {
       if (!this.fsatOperations.cost) {
         this.fsatOperations.cost = this.settings.electricityCost;
       }
       this.operationsForm = this.operationsService.getFormFromObj(this.fsatOperations);
+      this.setCo2SavingsData();
```
Comment (rmroot) [inline src/app/fsat/operations/operations.component.ts:92]: Don't want to call `setCo2SavingsDate()` twice. Just keep the second one

File: src/app/shared/models/fans.ts [hunk 12]
```diff
@@ -351,6 +355,7 @@ export interface FsatInput {
   //modified, optimal
   fanEfficiency?: number;
   fanType?: number;
+  cO2SavingsData?: Co2SavingsData;
```
Comment (rmroot) [inline src/app/shared/models/fans.ts:358]: don't need `cO2SavingsData` on `FsatInput` object

File: src/app/shared/models/fans.ts [hunk 13]
```diff
@@ -497,3 +504,12 @@ export const fanChecklistQuestions =  {
   radialFanCleanAir: 'Radial fan handling clean air',
 };
 
+export interface ExploreOpportunitiesResults {
```
Comment (rmroot) [inline src/app/shared/models/fans.ts:507]: This interface is unused. Remove

**Review by rmroot — CHANGES_REQUESTED** [review]

**Review by rmroot — APPROVED** [review]

**Pull request merged**